### DISCUSSION TITLE
cannot access VHDX : ERROR_SHARING_VIOLATION

### DISCUSSION THREAD

**Discussion by tjarriaultcontractor** [microsoft/WSL#11717] [General]
Hi,
on Windows 10 or 11, i have often an issue:
Failure to attach disk "C:\devenv\windows\vdx\ext4.vhdx" to WSL2: The process cannot access the file because it is being used by another process.
Error code: Wsl/Service/CreateInstance/MountVhd/HCS/ERROR_SHARING_VIOLATION

Do you know why i have this error and how i can solve definitevely this error ?

Regards
Tony

**Comment by hausburo-QX** [comment]
THIS is definitely too late but i was looking for answer for i had the same problem today; the problem is that the  "extv4.vhdx" file is already is attached to windows even if the WSL thought it wasn't, when i started the terminal (In my case) WSL tried to reattach it again and failed ,hence : error_sharing violation, failed to attach disk, HERE's what you can do:
1-run power shell as administrator ,then type diskpart, you should get disk version , then type "list vdisk"(the quotes just indicate , don't actually write them in PowerShell terminal) ,then  it'll return vdisk 0 and disk 1 status "attached not open".
2-type "select disk1" ,then type detail disk , it should return "There is no virtual disk selected...etc"
3-lastly type "select vdisk file = (exactly copy the file starting with c:/ that was given when you wrote "list vdisk"), it'll return :"file selected successfully" then  type detach file , finally type "exit" , go on terminal and you're done
.
